nomad/structs/structs.go

```diff
@@ -2879,6 +2891,13 @@ func (n *NodeResources) Copy() *NodeResources {
 	newN.Cpu = n.Cpu.Copy()
 	newN.Networks = n.Networks.Copy()
 
+	if n.NodeNetworks != nil {
+		newN.NodeNetworks = make([]*NodeNetworkResource, len(n.NodeNetworks))
+		for i, nn := range n.NodeNetworks {
+			newN.NodeNetworks[i] = nn.Copy()
+		}
+	}
```

Comment:
@schmichael I noticed this field was missing from `Node.Copy()`, but I don't know if it would cause this problem since fingerprinting has been broken as well.

Reply:
Great catch regardless! (Reason 9001 why we need to autogenerate these Copy methods)